src/bpfprogram.cpp

```diff
@@ -63,24 +63,29 @@ void BpfProgram::set_attach_target(const Probe &probe,
 
   const std::string &mod = probe.path;
   const std::string &fun = probe.attach_point;
+  const std::string &attach_target = !mod.empty() ? mod + ":" + fun : fun;
```

Comment:
```suggestion
  const std::string attach_target = !mod.empty() ? mod + ":" + fun : fun;
```

Reply:
Here, I think the ref makes sense as for the false case (`mod.empty()`), it doesn't have to copy `fun`.

Reply:
I wonder how the compiler handles this. I think because the type is either a new string or a ref it might have to default to making a copy (non ref) because it doesn't know at compile time.

Reply:
Yeah, it probably does that. Let me change to non-ref.